### DISCUSSION TITLE
mbtile 'loads' in Project but is not visible in QField on iOS device

### DISCUSSION THREAD

**Discussion by martingeh** [opengisch/QField#3789] [Q&A]
Morning all,
New to this forum and have tried many times to be able to see the .mbtiles created in QGis.
It has uploaded to QFieldCloud without issue and is only 26.9MB.  I've checked in QFieldCloud online and the file is there, but isn't being displayed.
The file has been selected as the Base Map in the Project Properties.
This file is necessary as field work will be done with no internet/cell coverage.

Any suggestions would be greatly appreciated.
![No mbtile displayed](https://user-images.githubusercontent.com/80141340/210294148-5c4d2b57-3a59-4d5d-bdf3-f42e4a5a4aaf.jpeg)
![Project files including mbtiles in QFieldCloud](https://user-images.githubusercontent.com/80141340/210294153-b5708303-d087-4118-b06a-fcf9c8090435.jpeg)

**Comment by martingeh** [comment]
Well I haven't had much luck in getting the images to show in Qfield when out of mobile (cell) coverage.  After my issue yesterday was resolved, I applied that check to this process as well and now all I see is a portion of the edge of the image.  
From this I can see that the image is where is should be, but it simply won't load.  The file is 182MB - is that too big?  Should I be making them smaller and if so, what is the optimum size per image?

Thanks in advance.

![IMG_4583](https://user-images.githubusercontent.com/80141340/215362903-78118ce9-6429-4cee-b6e1-f408106f4b1c.jpeg)

**Reply by nirvn** [reply]
@martingeh , try to attach a reduced sample project here, it'll be easier to help you out.

**Reply by martingeh** [reply]
Okay.  I'll get on to that and post when done.  Thank you.

**Reply by martingeh** [reply]
Hi Mathieu 
I've compressed and attach the project file.
[Erosion.zip](https://github.com/opengisch/QField/files/10531583/Erosion.zip)

**Reply by nirvn** [reply]
@martingeh , opening this project on my Android device using 2.6.2, I see the mbtiles are rendering without problem. Have you tried this specific sample project on your iOS device to confirm it replicates the issue? If so, it might be iOS-specific here.

**Reply by martingeh** [reply]
Yes, I use iOS device and have tried this sample project and a range of other variations, all without the mbtiles rendering as expected.  They do render on OSX though.

**Reply by nirvn** [reply]
@martingeh , do you happen to have an Android phone you could test it against?

**Reply by martingeh** [reply]
I don't, but I'll ask someone this afternoon to try it for me.  Should I log an issue for this?

**Reply by nirvn** [reply]
@martingeh , opened an issue already.

**Reply by martingeh** [reply]
Okay.  Thanks

**Reply by martingeh** [reply]
HI Mathieu.  I have tried this on an Android unit and it does show what I'm expecting.  I'll just have to work without a raster file while off line/out of cell coverage for now.  Thanks again for your help - much appreciated.

**Reply by nirvn** [reply]
@martingeh , actually, there's a known issue on iOS whereas JPG datasets aren't rendered. I've just checked your mbtiles, and the tile format is JPEG. You could try to create a mbtiles using PNG8 for the tile. It'll result in bigger GPKGs but it might very well unlock it for iOS.

**Reply by nirvn** [reply]
@martingeh , alternatively, you could try alternative formats with overview pyramids (see https://www.opengis.ch/2020/06/09/offline-wms-benchmarking-raster-formats-for-qfield/ , lots of useful info in there).

**Reply by nirvn** [reply]
@martingeh , actually, I'm 100% sure it'll work with mbtiles if you set the tile format to PNG.

The little you see rendered (i.e. the flipped L) are individual tiles saved in PNG format due to need for an alpha channel as the region doesn't fully cover the tile:

![image](https://user-images.githubusercontent.com/1728657/215430955-a7925ed4-f895-4c71-a752-21100a32272a.png)

So you've got your solution.

**Reply by martingeh** [reply]
Legend!  I'll work on that tonight and let you know how I go.

**Reply by martingeh** [reply]
Fantastic!!! Thankyou very much Mathieu.

**Comment by nirvn** [comment]
QField on iOS doesn't render JPG datasets (or in this case mbtiles with individual tiles being JPG). 

The issue is tracked here: https://github.com/opengisch/QField/issues/3611
